### PR TITLE
Make system version & company info visibility through the guest API configurable

### PR DESCRIPTION
**Pull request by Anuril**
Discord-User Pielgrzym mentioned that he'd like to hide the version number from the public. 

I also found that the api/guest/system/company endpoint returned the company data. 

AFAIK only the version endpoint is used in the admin area. The company endpoint isn't - and to be honest, I also like the idea of being able to disable this. 

I've added the settings to Admin -> Company, and added checks to getVersion & getCompany. 

Version now checks if the request comes from a logged in Admin, and returns the version (no matter the setting). 
After that, both functions check for the setting in the database.

### CODE REVIEW

**Review by jaapmarcus — APPROVED** [review]

File: src/modules/System/Api/Guest.php [hunk 1]
```diff
@@ -23,6 +23,19 @@ class Guest extends \Api_Abstract
...
     public function version()
     {
+        // check if the user is logged in as admin and if so, return the version
+        if ($this->di['auth']->isAdminLoggedIn()) {
+            return $this->getService()->getVersion();
+        }
+
+        // check if the "show_version_public" parameter is set to true
+        $showVersionPublic = $this->getService()->getParamValue('show_version_public');
+        if ($showVersionPublic == 1) {
```
Comment (BelleNottelling) [inline src/modules/System/Api/Guest.php:33]: Don't need to check and verify if it's set to 1.
`if($showVersionPublic)` will evaluate to true if the value is set and non-falsey. Meaning if it's set and is equal to 1, the check will pass. 

If it's not set or it's set to 0, the check will not pass.

File: src/modules/System/Api/Guest.php [hunk 2]
```diff
@@ -33,7 +46,14 @@ public function version()
      */
     public function company()
     {
-        return $this->getService()->getCompany();
+        // check if the "show_company_public" parameter is set to true
+        $showCompanyPublic = $this->getService()->getParamValue('show_company_public');
+        if ($showCompanyPublic == 1) {
```
Comment (BelleNottelling) [inline src/modules/System/Api/Guest.php:51]: Same thing here

**Review by BelleNottelling — CHANGES_REQUESTED** [review]
This simply just won't work. Lots of portions of the front-end use the `company` API endpoint and will be adversely effected by removing or disabling that endpoint. https://github.com/search?q=repo%3AFOSSBilling%2FFOSSBilling%20guest.system_&type=code

**Comment by Anuril** [comment]
You're right... I missed that somehow :O 

I still don't like having my VAT, Bank Account, Phone number and E-Mail publicly accessible via an API. 

I want to rework this but I'll do it at a later stage. 

Sorry for wasting your time :/

**Comment by BelleNottelling** [comment]
> You're right... I missed that somehow :O
> 
> I still don't like having my VAT, Bank Account, Phone number and E-Mail publicly accessible via an API.
> 
> I want to rework this but I'll do it at a later stage.
> 
> Sorry for wasting your time :/

My suggestion is to just `unset` the array keys you want to keep a little more private if the API request is being made by someone who's not authenticated.

```PHP
public function company()
{
    $companyInfo = $this->getService()->getCompany();
    $auth = $this->di['auth'];
    $showCompanyPublic = $this->getService()->getParamValue('show_company_public');
    
    if(!$auth->isAdminLoggedIn() && !$auth->isClientLoggedIn() && !$showCompanyPublic){
        unset($companyInfo['vat_number']);
        unset($companyInfo['email']);
        unset($companyInfo['tel']);
        unset($companyInfo['account_number']);
        unset($companyInfo['number']);
    
        unset($companyInfo['address_1']);
        unset($companyInfo['address_2']);
        unset($companyInfo['address_3']);
    }
    
    return $companyInfo;
}
```

**Review by BelleNottelling — APPROVED** [review]
I've made some changes and this PR is now to a point where I'm personally happy with it

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**